storage/tables/data-types/index.md

```diff
@@ -993,29 +993,111 @@ The `TIMESTAMP` base type represents a date value with a time portion.
-  - Other components will create non-typed table, where all columns have text type.
+Table with native datatypes is labeled in UI with a badge:
 
-   **_NOTE:_**  Not all database extractors and transformations can produce basetypes.
+{: .image-popup}
+![Screenshot - Table with native datatypes](/storage/tables/data-types/typed-table.png)
 
-   **_NOTE:_**  Some components may produce basetypes for columns and thus produce typed tables
+### How to create a typed table?
 
-#### Pros and cons
-- **Pros**
-  - Manipulation with such data is easier and un/loading can be faster.
-  - There is no need for casting when using [Read-only IM](/storage/backends/byodb/#read-only-access-to-project-storage)
-- **Cons**
-  - Datatypes in typed tables are given and there is no way how it can be altered (neither UI nor API)
+#### Manually via API
 
-Table with native datatypes is labeled in UI with a badge:
+Table with type definition is created using [tables-definition enpoint](https://keboola.docs.apiary.io/#reference/tables/create-table-definition/create-new-table-definition) and data is then loaded into it. Datatypes used in this endpoint have to correspond with the Storage backend which your project uses. Alternatively, you can use [BASETYPES](#base-types).
 
-{: .image-popup}
-![Screenshot - Table with native datatypes](/storage/tables/data-types/typed-table.png)
+#### Output mapping of a component
+
+A component may provide information about columns datatypes in its data manifest. Database extractors and transformations matching storage backend (e.g. Snowflake SQL transformation on Snowflake storage backend) will create storage tables with same types. Database extractors and transformations NOT matching backend will create storage tables using [BASETYPES](#base-types). 
+
+For example, this is how you can create typed tables in Snowflake SQL transformation, that will be imported to Storage as typed tables: 
+
+```sql
+-- create a table with datatypes
+CREATE OR REPLACE TABLE "typed_table" (
+    "id" NUMBER,
+    "name" VARCHAR(255),
+    "created_at" TIMESTAMP_NTZ
+);
+
+-- insert some data
+INSERT INTO "typed_table"
+VALUES
+    (1, '75', '2020-01-01 00:00:00');
+
+-- create a table with different datatypes from an existing table
+CREATE OR REPLACE TABLE "typed_table_2" AS
+SELECT
+    "id"::varchar AS "string_id",
+    "name"::number AS "numeric_name",
+    "created_at"::date AS "typed_date"
+FROM
+    "typed_table";
+```
+
+***Note:** The datatype hinting is components' responsibility, so all components need to be updated by their respective authors to support this. This is why most components don't produce basetypes for columns yet and thus produce tables without types. Not even all database extractors and transformations produce basetypes yet. There is no list of components that support this feature. You may check the component's documentation to see if it supports native datatypes.* 

```

Comment:
To je sice pravda, ale pro usera dost matoucín navíc. "so all components need to be updated by their respective authors to support this." se nikdy nestane a spousta komponent bude navždy produkovat stringy.

Providing information about data types is the responsibility of each component. So far typed tables are created (provided that the project feature is enabled) by Snowflake transformations, and some Database Extractors (Snowflake MySQL/Postgre/MSSQL (pokud je to pravda)). There is no list of components that support this feature. You may check the component's documentation to see if it supports native datatypes.